hivemind_listener/protocol.py

```
@@ -176,18 +175,87 @@ class PluginOptions:
 
-class AudioBinaryProtocol:
+@dataclass
+class AudioBinaryProtocol(BinaryDataHandlerProtocol):
     """wrapper for encapsulating logic for handling incoming binary data"""
     plugins: Optional[PluginOptions] = None
     utterance_transformers: Optional[UtteranceTransformersService] = None
     metadata_transformers: Optional[MetadataTransformersService] = None
     dialog_transformers: Optional[DialogTransformersService] = None
-
+    config: Dict[str, Any] = field(default_factory=dict)
     hm_protocol: Optional['AudioReceiverProtocol'] = None
```

Ruff (0.8.2)

Undefined name `AudioReceiverProtocol`

(F821)